perfkitbenchmarker/cloudstack/cloudstack_virtual_machine.py

```diff
@@ -0,0 +1,217 @@
+# Copyright 2014 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from perfkitbenchmarker import flags
+from perfkitbenchmarker import linux_virtual_machine as linux_vm
+from perfkitbenchmarker import virtual_machine
+from perfkitbenchmarker import vm_util
+from perfkitbenchmarker.cloudstack import cloudstack_network
+from perfkitbenchmarker.cloudstack import cloudstack_disk
+from perfkitbenchmarker.cloudstack import util
+
+import logging
+
+
+UBUNTU_IMAGE = 'Ubuntu 14.04.2 HVM base (64bit)'
+RHEL_IMAGE = 'CentOS 7 HVM base (64bit)'
+
+FLAGS = flags.FLAGS
+
+
+class CloudStackVirtualMachine(virtual_machine.BaseVirtualMachine):
+  """Object representing a CloudStack Virtual Machine."""
+
+  DEFAULT_ZONE = 'QC-1'
+  DEFAULT_MACHINE_TYPE = '1vCPU.1GB'
+  DEFAULT_IMAGE = 'Ubuntu 14.04.2 HVM base (64bit)'
+  DEFAULT_USER_NAME = 'cca-user'
+  DEFAULT_PROJECT = 'cloudops-Engineering'
+
+
+  def __init__(self, vm_spec):
+    """Initialize a CloudStack virtual machine.
+
+    Args:
+      vm_spec: virtual_machine.BaseVirtualMachineSpec object of the vm.
+    """
+    super(CloudStackVirtualMachine, self).__init__(vm_spec)
+
+    self.network = \
```

Comment:
will need to change when you pull in the latest updates (networks and firewalls now get passed in through __init__)